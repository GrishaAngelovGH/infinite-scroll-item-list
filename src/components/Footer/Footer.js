import { useSelector } from 'react-redux'

import { StarFilled, TagsFilled } from '@ant-design/icons'
import { Tooltip } from 'antd'

import './Footer.scss'

const Footer = () => {
  const items = useSelector((state) => Object.values(state.items))
  const favorites = items.filter(v => v.isFavorite)

  return (
    <div className='footer'>
      <Tooltip title='Fetched Items' placement='left'>
        <h1><TagsFilled style={{ fontSize: 30, paddingRight: 10 }} /> {items.length}</h1>
      </Tooltip>

      <Tooltip title='Favorite Items' placement='right'>
        <h1><StarFilled style={{ fontSize: 30, paddingRight: 10 }} /> {favorites.length}</h1>
      </Tooltip>
    </div>
  )
}

export default Footer