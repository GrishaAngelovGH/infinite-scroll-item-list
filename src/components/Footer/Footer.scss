.footer {
  display: flex;
  justify-content: space-evenly;

  h1 {
    margin: 0;
    display: flex;
    align-items: center;
  } 
}
