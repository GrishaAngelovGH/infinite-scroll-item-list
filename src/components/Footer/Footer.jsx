import { useSelector } from 'react-redux'
import { selectItemValues, selectFavoriteItems } from 'selectors/itemRowsSelector'

import { StarFilled, TagsFilled } from '@ant-design/icons'
import { Tooltip } from 'antd'

import './Footer.scss'

const Footer = () => {
  const items = useSelector(selectItemValues)
  const favorites = useSelector(selectFavoriteItems)

  return (
    <div className='footer'>
      <Tooltip title='Fetched Items' placement='left'>
        <h1><TagsFilled style={{ fontSize: 30, paddingRight: 10 }} /> {items.length}</h1>
      </Tooltip>

      <Tooltip title='Favorite Items' placement='right'>
        <h1><StarFilled style={{ fontSize: 30, paddingRight: 10 }} /> {favorites.length}</h1>
      </Tooltip>
    </div>
  )
}

export default Footer