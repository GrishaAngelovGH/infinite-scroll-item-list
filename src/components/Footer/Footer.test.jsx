import { render } from '@testing-library/react'

import { Provider } from 'react-redux'

import Footer from './Footer'

test('should render Footer component', () => {
  const view = render(
    <Provider store={globalThis.mockStore}>
      <Footer />
    </Provider>
  )

  expect(view).toMatchSnapshot()
})