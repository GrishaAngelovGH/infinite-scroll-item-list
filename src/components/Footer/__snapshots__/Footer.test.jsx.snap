// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Footer component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="footer"
      >
        <h1
          aria-describedby="test-id"
        >
          <span
            aria-label="tags"
            class="anticon anticon-tags"
            role="img"
            style="font-size: 30px; padding-right: 10px;"
          >
            <svg
              aria-hidden="true"
              data-icon="tags"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M483.2 790.3L861.4 412c1.7-1.7 2.5-4 2.3-6.3l-25.5-301.4c-.7-7.8-6.8-13.9-14.6-14.6L522.2 64.3c-2.3-.2-4.7.6-6.3 2.3L137.7 444.8a8.03 8.03 0 000 11.3l334.2 334.2c3.1 3.2 8.2 3.2 11.3 0zm122.7-533.4c18.7-18.7 49.1-18.7 67.9 0 18.7 18.7 18.7 49.1 0 67.9-18.7 18.7-49.1 18.7-67.9 0-18.7-18.7-18.7-49.1 0-67.9zm283.8 282.9l-39.6-39.5a8.03 8.03 0 00-11.3 0l-362 361.3-237.6-237a8.03 8.03 0 00-11.3 0l-39.6 39.5a8.03 8.03 0 000 11.3l243.2 242.8 39.6 39.5c3.1 3.1 8.2 3.1 11.3 0l407.3-406.6c3.1-3.1 3.1-8.2 0-11.3z"
              />
            </svg>
          </span>
           
          6
        </h1>
        <h1
          aria-describedby="test-id"
        >
          <span
            aria-label="star"
            class="anticon anticon-star"
            role="img"
            style="font-size: 30px; padding-right: 10px;"
          >
            <svg
              aria-hidden="true"
              data-icon="star"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
              />
            </svg>
          </span>
           
          2
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="footer"
    >
      <h1
        aria-describedby="test-id"
      >
        <span
          aria-label="tags"
          class="anticon anticon-tags"
          role="img"
          style="font-size: 30px; padding-right: 10px;"
        >
          <svg
            aria-hidden="true"
            data-icon="tags"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M483.2 790.3L861.4 412c1.7-1.7 2.5-4 2.3-6.3l-25.5-301.4c-.7-7.8-6.8-13.9-14.6-14.6L522.2 64.3c-2.3-.2-4.7.6-6.3 2.3L137.7 444.8a8.03 8.03 0 000 11.3l334.2 334.2c3.1 3.2 8.2 3.2 11.3 0zm122.7-533.4c18.7-18.7 49.1-18.7 67.9 0 18.7 18.7 18.7 49.1 0 67.9-18.7 18.7-49.1 18.7-67.9 0-18.7-18.7-18.7-49.1 0-67.9zm283.8 282.9l-39.6-39.5a8.03 8.03 0 00-11.3 0l-362 361.3-237.6-237a8.03 8.03 0 00-11.3 0l-39.6 39.5a8.03 8.03 0 000 11.3l243.2 242.8 39.6 39.5c3.1 3.1 8.2 3.1 11.3 0l407.3-406.6c3.1-3.1 3.1-8.2 0-11.3z"
            />
          </svg>
        </span>
         
        6
      </h1>
      <h1
        aria-describedby="test-id"
      >
        <span
          aria-label="star"
          class="anticon anticon-star"
          role="img"
          style="font-size: 30px; padding-right: 10px;"
        >
          <svg
            aria-hidden="true"
            data-icon="star"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 00.6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0046.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
            />
          </svg>
        </span>
         
        2
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
